src/redux/ducks/locs/locModalActions.js

```diff
@@ -1,11 +1,10 @@
-import {showLOCModal, showIssueLHModal, showUploadedFileModal } from '../../../redux/ducks/ui/modal';
-import {storeLOCAction} from '../../../redux/ducks/locs/loc';
+import {showLOCModal, showIssueLHModal, showUploadedFileModal } from '../ui/modal';
+import {storeLOCAction} from './loc';
 import IPFSDAO from '../../../dao/IPFSDAO';
 
 export const handleShowLOCModal = (loc) => (dispatch) => {
     dispatch(storeLOCAction(loc));
-    const locExists = !!loc;
-    dispatch(showLOCModal({locExists}));
+    dispatch(showLOCModal({locExists: !!loc}));
 };
 
 export const handleShowIssueLHModal = (loc) => (dispatch) => {
```

Comment:
Use IPFSDAO.getNode() getter below, we can implement mock functionality inside of it by example with OrbitDAO for testing purposes.

Also don't access to the immutable record properties through the .get() - add there additional getters, so we'll have autocomplete (and less chance of making a mistake with it).

And use type hinting for your models. For instance
`export const handleViewContract = (loc: LOCModel)`